File: Lab_10/BankApp/Models/Account.cs
namespace BankApp.Models;

public class Account
{
    public string AccountNumber { get; set; }
    public decimal Balance { get; set; }
    public List<Transaction> Transactions { get; }

    public Account(string accountNumber, decimal initialDeposit)
    {
        this.AccountNumber = accountNumber;
        this.Balance = initialDeposit;
        Transactions = new List<Transaction>
        {
            new Transaction("Deposit", initialDeposit)
        };
    }

    public void Deposit(decimal amount)
    {
        
        Balance += amount;
        Transactions.Add(new Transaction("deposit",amount));
        Console.WriteLine($"Deposited {amount} into {AccountNumber}. New balance: {Balance}");
    }

    public void Withdraw(decimal amount)
    {
        if (amount < Balance)
        {
            Balance -= amount;
            Transactions.Add(new Transaction("Withdraw", amount));
            Console.WriteLine($"Withdrawn {amount} from {AccountNumber}. New balance {Balance}");
        }
        else
        {
            Console.WriteLine("Insufficient Funds");
        }
        
    }
}

File: lab10/Bank Account Command Pattern/BankApp/src/Models/Account.java
package Models;

import java.util.ArrayList;
import java.util.List;

public class Account {
    private String accountNumber;
    private double balance;
    private List<Transaction> transactions;
    public Account(String accountNumber, double initialDeposit) {
        this.accountNumber = accountNumber;
        this.balance = initialDeposit;
        this.transactions = new ArrayList<>();
        transactions.add(new Transaction("Deposit", initialDeposit));

    }

    public String getAccountNumber() {
        return accountNumber;
    }
    public double getBalance() {
        return balance;
    }
    public List<Transaction> getTransactions() {
        return transactions;
    }
    public void deposit(double amount) {
        this.balance += amount;
        transactions.add(new Transaction("Deposit", amount));
        System.out.println("Deposited " + amount + " into " + accountNumber + ". New balance: " + balance);

    }
    public void withdraw(double amount) {
        if(amount <= balance) {
            balance -= amount;
            transactions.add(new Transaction("Withdraw", amount));
            System.out.println("Withdrawn " + amount + " from " + accountNumber + ". New balance: " + balance);
        }
        else{
            System.out.println("Insufficient balance");
        }
    }
}
